File: gripper_minimal_service/scripts/gripper_minimal_client_v2.py
#!/usr/bin/python3

import rclpy
from rclpy.node import Node
from hrim_actuator_gripper_srvs.srv import ControlFinger


class MinimalClient(Node):

    def __init__(self):
        super().__init__('mara_minimal_client')

        # Create a client for service "/hrim_actuation_gripper_000000000004/goal"
        self.client = self.create_client(ControlFinger, "/hrim_actuation_gripper_000000000004/goal")

        # Wait for service to be avaiable before calling it
        while not self.client.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('service not available, waiting again...')

        # Create request with the same type as the service, ControlFinger
        self.req = ControlFinger.Request()

    def send_request(self):
        self.req.goal_linearposition = 0.87
        self.future = self.client.call_async(self.req)


def main(args=None):
    rclpy.init(args=args)

    node = MinimalClient()

    # Call service and spin
    node.send_request()
    rclpy.spin_until_future_complete(node, node.future)

    # Analyze the result
    if node.future.result() is not None:
        node.get_logger().info('Goal accepted: %d: ' % node.future.result().goal_accepted)
    else:
        node.get_logger().error('Exception while calling service: %r' % node.future.exception())

    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: gripper_minimal_service/src/gripper_minimal_client_v2.cpp
#include "rclcpp/rclcpp.hpp"
#include "hrim_actuator_gripper_srvs/srv/control_finger.hpp"
#include <chrono>

using namespace std::chrono_literals;

class MinimalClient : public rclcpp::Node
{
  public:
    MinimalClient(): Node("gripper_minimal_client")
    {
      client = this->create_client<hrim_actuator_gripper_srvs::srv::ControlFinger>(
        "/hrim_actuator_gripper_000000000004/fingercontrol");

      // Wait for service to be avaiable before calling it
      while (!client->wait_for_service(1s)) {
        RCLCPP_INFO(this->get_logger(), "Service not available, waiting again...");
      }

      request = std::make_shared<hrim_actuator_gripper_srvs::srv::ControlFinger::Request>();
    }

    void send_request();
    rclcpp::Client<hrim_actuator_gripper_srvs::srv::ControlFinger>::SharedFuture future;

  private:
    rclcpp::Client<hrim_actuator_gripper_srvs::srv::ControlFinger>::SharedPtr client;
    rclcpp::Client<hrim_actuator_gripper_srvs::srv::ControlFinger>::SharedRequest request;
};

void MinimalClient::send_request(){
  request->goal_angularposition = 0.;;
  this->future = client->async_send_request(request);
}


int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);

  auto node = std::make_shared<MinimalClient>();

  node->send_request();
  rclcpp::spin_until_future_complete(node, node->future);

  // Analyze the result
  auto result = node->future.get();
  if (result) {
    RCLCPP_INFO(node->get_logger(), "Result of gripper_minimal_client: %d", result->goal_accepted);
  } else {
    RCLCPP_ERROR(node->get_logger(), "Interrupted while waiting for response. Exiting.");
  }

  rclcpp::shutdown();
  return 0;
}
